firmware/shared/src/io/io_fileSystem.h

```diff
@@ -0,0 +1,28 @@
+#include <stdint.h>
+
+typedef enum
+{
+    FILE_OK    = 0,
+    FILE_ERROR = -1, // general error
+} FileSystemStatus;
+
+// init the file system
+// config the file system
+// mount the file system
+int io_fileSystem_init(void);
+
+// open a file and return the file descriptor
+int io_fileSystem_open(const char *path);
+
+// write return error code
+int io_fileSystem_write(int fd, void *buf, size_t size);
+
+// given fd, buf, size, return error code
+int io_fileSystem_read(int fd, void *buf, size_t size);
+// close file
+int io_fileSystem_close(int fd);
+
+uint32_t io_fileSystem_getBootCount(void);
+
+// do the concrete write operation to the hardware
+int io_fileSystem_sync(int fd);
```

Comment:
I really like this abstraction layer idea, nice! I'm glad we're keeping little FS around since it is a much better general purpose filesystem.

One concern, is we're destroying a lot of information returning just FILE_OK and FILE_ERROR (i.e., lots of other more specific error codes that will all get mapped to FILE_ERROR). Maybe just return the error code from the filesystem specific functions as an int, and its up to the user of this to interpret/log them later? What do you think?